greeting.rb

```diff
@@ -0,0 +1,42 @@
+class Greeting
+  def greet(*name)
+    if name.any?
+      low_case = name.reject { |x| x == x.upcase }
+      up_case= name.select { |r| r == r.upcase }
+      up_case_string = up_case.join(",")
+      name_string = name.join("")
+      if name.length == 1 and name == low_case
+        "Hello, #{name.join(",")}."
+      elsif name.length == 1 and name == up_case
+        "HELLO #{name.join(",")}!"
+      elsif name.length == 2 and !name_string.include?(",") and !name_string.include?("\"")
+        "Hello, #{name.first} and #{name.last}."
+      elsif name.length > 2 and name == low_case
+        "Hello, #{name[0..-2].join(", ")}, and #{name.last}."
+      elsif name.length > 2 and name.include? up_case_string
+        "Hello, #{low_case[0..-2].join(', ')} and #{low_case.last}. AND HELLO #{up_case.join('')}!"
+      elsif name.length == 2 and name_string.include?(",") and  !name_string.include?("\"")
+        separated_names = split_names(name)
+        "Hello, #{separated_names[0..-2].join(', ')}, and#{separated_names.last}."
+      elsif name.length == 2 and name_string.include?("\"")
+        converted_names = escape_commas(name)
+        "Hello, #{converted_names.first} and #{converted_names[1..-2].join("")},#{converted_names.last}."
+      end
```

Comment:
There is some duplication going on in the if statements which I think might actually be easier to read if you used nested if statements. So basically you'd have an outside 3 if statements (`name.length == 1`, `name.length == 2`, and `name.length > 2`. And then you'd have inner if statements that would check if it was all upper case/lower case, etc.